src/test/spec/versioned_api.rs

```diff
@@ -0,0 +1,67 @@
+use tokio::sync::{RwLockReadGuard, RwLockWriteGuard};
+
+use crate::{
+    bson::doc,
+    options::{ServerApi, ServerApiVersion},
+    test::{run_spec_test, EventClient, TestClient, CLIENT_OPTIONS, LOCK},
+};
+
+use super::run_unified_format_test;
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn run() {
+    let _guard: RwLockWriteGuard<_> = LOCK.run_exclusively().await;
+    // TODO RUST-725 Unskip these tests on 5.0
+    if TestClient::new().await.server_version_gte(5, 0) {
+        return;
+    }
+    run_spec_test(&["versioned-api"], run_unified_format_test).await;
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn transaction_handling() {
```

Comment:
This is the prose version of a spec test right? If so, can we add a comment indicating that along with a TODO with the ticket to convert transactions to the unified runner (RUST-817, or maybe RUST-734)?

Reply:
oh good catch I meant to do this but must've forgotten, fixed